packages/tizen-remote/test/e2e/tizen-remote.e2e.spec.js

```diff
@@ -0,0 +1,364 @@
+ *
+ * If the above env variables are _not_ present, a mock server will be started on a random port.
+ *
+ * This is optional:
+ *
+ * - `TEST_TIZEN_REMOTE_TOKEN`: The token to use for the WebSocket server
+ *
+ * Tokens are associated with "names", and the default name in this module is `Appium`.
+ *
+ * @module
+ */
+
+import {createSandbox} from 'sinon';
+import getPort from 'get-port';
+import {TestWSServer} from './server';
+import unexpected from 'unexpected';
+import {constants, Event, Keys, TizenRemote} from '../../lib/index';
+import d from 'debug';
+import unexpectedSinon from 'unexpected-sinon';
+import unexpectedEventEmitter from 'unexpected-eventemitter';
+import {Env} from '@humanwhocodes/env';
+
+const env = new Env();
+
+const HOST = /** @type {string} */ (env.get('TEST_TIZEN_REMOTE_HOST', '127.0.0.1'));
+const PORT = env.get('TEST_TIZEN_REMOTE_PORT');
+const TOKEN = env.get('TEST_TIZEN_REMOTE_TOKEN');
+
+const debug = d('tizen-remote:test:e2e:ws');
+
+const expect = unexpected.clone().use(unexpectedSinon).use(unexpectedEventEmitter);
+
+describe('websocket behavior', function () {
+  /** @type {TestWSServer} */
+  let server;
+
+  /** @type {number} */
+  let port;
+
+  /** @type {TizenRemote} */
+  let remote;
+
+  /** @type {sinon.SinonSandbox} */
+  let sandbox;
+
+  /** @type {import('../../lib').TizenRemoteOptions} */
+  let remoteOpts;
+
+  /** @type {string} */
+  let token;
+
+  before(async function () {
+    if (PORT) {
+      this.timeout('1m');
+      port = Number(PORT);
+      if (TOKEN) {
+        token = TOKEN;
+      } else {
+        debug('[SETUP] Getting token from %s:%d; this may be slow...', HOST, port);
+        const remote = new TizenRemote({
+          host: HOST,
+          port
+        });
+        token = await new Promise((resolve, reject) => {
+          remote
+            .once(Event.TOKEN, (t) => {
+              resolve(t);
+            })
+            .on('error', reject);
+          remote.connect();
+        });
+        debug('[SETUP] Got token: %s', token);
+        await remote.disconnect();
+      }
+    } else {
+      port = await getPort();
+    }
+  });
+
+  beforeEach(function () {
+    sandbox = createSandbox();
+    if (!PORT) {
+      server = new TestWSServer({
+        host: HOST,
+        port,
+        path: constants.API_PATH_V2,
+      });
+    }
+    remoteOpts = {
+      host: HOST,
+      port,
+      token
+    };
+  });
+
+  afterEach(async function () {
+    sandbox.restore();
+    if (remote?.isConnected) {
+      try {
+        debug('[CLEANUP] Disconnecting from remote attached to %s:%d', HOST, port);
+        await remote.disconnect();
+      } catch {}
+    }
+    if (server) {
+      debug('[CLEANUP] Stopping server listening on %s:%d', HOST, port);
+      try {
+        await server.stop();
+      } catch {}
+    }
+  });
+
+  it('should connect', async function () {
+    remote = new TizenRemote(remoteOpts);
+    remote.once(Event.TOKEN, (t) => {
```

Comment:
would be nice to have a non-event interface so we can `const token = await remote.getToken()` or something

Reply:
sarcasm is hard to read in text.  but anyway, yeah I think we don't even need this here and we should get it up there in `before()`

Reply:
oh, i wasn't being sarcastic. i had just forgotten that we already did have literally that 🤣